packages/theme/src/cli/services/dev.ts

```diff
@@ -81,12 +81,20 @@ export async function dev(options: DevOptions) {
   const port = options.port || String(await getAvailableTCPPort(Number(DEFAULT_PORT)))
 
   const storefrontPassword = await storefrontPasswordPromise
+
   const session = await initializeDevServerSession(
     options.theme.id.toString(),
     options.adminSession,
     options.password,
     storefrontPassword,
+    {
+      ...options.adminSession,
+      storefrontToken: options.storefrontToken,
```

Comment:
Yeah, it's hard to find balance between perf and readability but I agree this might be a case for readability. I made this PR to fix a bug in the first place, perf was just a side effect. Since the bug was fixed in a different way, I wasn't so sure the change was worth it for the perf alone, hence why I waited until your review. I'll close this for now 👍